### ISSUE TITLE
(Very) obscure words need culling

### ISSUE THREAD

**Issue by theoldbloke** [mozilla/DeepSpeech#3479]
I ran DeepSpeech on a very ordinary audio file (only 2 minutes) with a USA male English speaker.

It came up with some (very) obscure words. Here they are, with actual text following in brackets:

_philippowanians_ (Philip, you know)
_cortwrite_ (sort of written)
_mariannakookaland_ (week, marking a week)
_lilybean_ (likely to)

Take "_mariannakookaland_". It appears in _Punch_, January 17, 1891 and was never heard of again. Google returns a trifling 6 results, and only one is relevant - the same _Punch_ article on Project Gutenberg!

Apart from any Philippowanians from Mariannakookaland, this is definitely a case of more data not necessarily being an improvement.

From my untutored perspective it seems simple enough to filter the dictionaries against a word frequency list and discard the (very) obscure and bizarre.

More understandably, DeepSpeech also tripped up on some currently topical words:

_facino, fouchet_ (Fauci)
_bidden_ (Biden)
_supersede_ (super-spreader)

Thank you for DeepSpeech,

_**Mike**_

**Comment by chmodsss** [comment]
Yes, I too had some weird words. Doesn't Deepspeech contain a language model when predicting the words?. if so why do you think such words have been recognized. As in a generic English LM, these words must have had a very tiny probability right.

**Comment by reuben** [comment]
Instructions on how our language model is built are available here: https://deepspeech.readthedocs.io/en/latest/Scorer.html#reproducing-our-external-scorer

You should be able to build modified versions of the scorer with your own dataset or pruning parameters following those.
